core/consensus/babe/impl/block_executor.hpp

```diff
@@ -44,39 +41,48 @@ namespace kagome::consensus {
-    void processNextBlock(
-        const libp2p::peer::PeerId &peer_id,
-        const primitives::BlockHeader &header,
-        const std::function<void(const primitives::BlockHeader &)>
-            &new_block_handler);
+    //    /**
+    //     * Processes next header: if header is observed first it is added to
+    //     the
+    //     * storage, handler is invoked. Synchronization of blocks between new
+    //     one
+    //     * and the current best one is launched if required
+    //     * @param header new header that we received and trying to process
+    //     * @param new_block_handler invoked on new header if it is observed
+    //     first
+    //     * time
+    //     */
+    //    void processNextBlock(
+    //        const libp2p::peer::PeerId &peer_id,
+    //        const primitives::BlockHeader &header,
+    //        const std::function<void(const primitives::BlockHeader &)>
+    //            &new_block_handler);
+    //
+    //    /**
+    //     * Synchronize all missing blocks between the last finalized and the
+    //     new one
+    //     * @param new_header header defining new block
+    //     * @param next action after the sync is done
+    //     */
+    //    void requestBlocks(const libp2p::peer::PeerId &peer_id,
+    //                       const primitives::BlockHeader &new_header,
+    //                       std::function<void(outcome::result<void>)> &&next);
+    //
+    //    /**
+    //     * Synchronize all missing blocks between provided blocks (from and
+    //     to)
+    //     * @param from starting block of syncing blocks
+    //     * @param to last block of syncing block
+    //     * @param on_retrieved action after the sync is done
+    //     */
+    //    void requestBlocks(const primitives::BlockHash &from,
+    //                       const primitives::BlockHash &to,
+    //                       const libp2p::peer::PeerId &peer_id,
+    //                       std::function<void(outcome::result<void>)>
+    //                       &&on_retrieved);
 
-    /**
-     * Synchronize all missing blocks between the last finalized and the new one
-     * @param new_header header defining new block
-     * @param next action after the sync is done
-     */
-    void requestBlocks(const libp2p::peer::PeerId &peer_id,
-                       const primitives::BlockHeader &new_header,
-                       std::function<void()> &&next);
-
-    /**
-     * Synchronize all missing blocks between provided blocks (from and to)
-     * @param from starting block of syncing blocks
-     * @param to last block of syncing block
-     * @param on_retrieved action after the sync is done
-     */
-    void requestBlocks(const primitives::BlockHash &from,
-                       const primitives::BlockHash &to,
-                       const libp2p::peer::PeerId &peer_id,
-                       std::function<void()> &&on_retrieved);
+    // should only be invoked when parent of block exists
```

Comment:
I assume it'd be good to explain here also what happens otherwise. And maybe even elaborate a bit what's meant by 'exist'. I assume it means the parent should be in our block tree?

Reply:
It is not actual now. This check is added into applyBlock method